_posts/2021-04-27-a-gentle-introduction-to-iceberg.md

```diff
@@ -0,0 +1,276 @@
+organization due to Hive’s number of unwritten assumptions, the Hive connector 
+is the most popular connector in use for Trino. Virtually every big data query 
+engine uses the Hive model today in some form and so many companies use it to 
+store data in their data lakes. 
+
+So how did something with no specification and such a barrier to use become so 
+ubiquitous in data lakes? Well first, good marketing and all the false promises
+of big data, but second is that Hive didn’t get everything wrong. In fact, 
+without Hive, there may not have been a unified specification at all. Despite 
+the many hours data engineers have spent bashing their heads against the wall 
+with all the unintended consequences of Hive, it still served a very useful 
+purpose.
+
+So why did I just rant about Hive for so long if I’m here to tell you about 
+[Apache Iceberg](https://iceberg.apache.org/)? It’s impossible for a teenager 
+growing up today to truly appreciate music streaming services without knowing 
+what it was like to have an iPod with limited storage, or listening to a 
+scratched burnt CD that skips, or flipping your tape or record to side-B. The 
+same way anyone born before the turn of the millennium really appreciates 
+streaming services, so you too will appreciate Iceberg once you’ve learned the 
+intricacies of managing a data lake built on Hive and Hadoop. 
+
+If you haven’t used Hive before, this blog will outline just a few pain points 
+that come from this query engine to give you proper context. If you have already
+lived through these headaches, this blog will act as a guide to Iceberg from 
+Hive. This blog is the first in a series of blogs discussing Apache Iceberg in 
+great detail, through the lens of the Trino query engine user. If you’re not 
+aware of Trino (formerly PrestoSQL) yet, it is the project that houses the 
+founding Presto community after the 
+[founders of Presto left Facebook](https://trino.io/blog/2020/12/27/announcing-trino.html).
+This and the next couple of blogs will discuss the Iceberg specification and all
+the features Iceberg has to offer, many times in comparison with Hive. 
+
+Before jumping into the comparisons, what is Iceberg exactly? The first thing to
+understand is that Iceberg is not a file format, but a table format. It may not
+be clear what this means by just stating that, but the function of a table 
+format will become clearer as the improvements Iceberg brings from the Hive 
+table standard materialize. Iceberg doesn’t replace file formats like ORC and 
+Parquet, but is the layer between the query engine and the data. Iceberg maps 
+and indexes the files in order to provide a higher level abstraction that 
+handles the relational table format for data lakes. You will understand more 
+about table formats through example as you read along in throughout this series.
+
+Hidden Partitions
+Since most developers' interactions with the table format is via the query 
```

Comment:
The first sentence you suggested here is incomplete. I think I get what you're going for though..it's a long sentence.

Reply:
Since most developers and users interact with the table format is via the query  language, a noticeable difference is the flexibility you have while creating a  partitioned table.